Automattic/wp-calypso
# Plans, Activity Log: Remove activity-log/display-rules and jetpack/only-realtime-products flags

Resolves `1200412004370260-as-1201266448248953`.

This PR removes both the `activity-log/display-rules` and `jetpack/only-realtime-products` feature flags from the Calypso code base, setting the app to behave as if these flags were permanently set to `true`.

#### Changes proposed in this Pull Request

* Additions:
  * Add CSS class names to the `display-price` component that correspond to the type of pricing to be rendered (e.g., `is-free`, `is-deprecated`).

* Removals:
  * Remove the `activity-log/display-rules` feature flag from all environment configs.
  * Remove the `jetpack/only-realtime-products` feature flag from all environment configs.
  * Remove the `ONLY_REALTIME_PRODUCTS` entry from the list of CRO iterations.
  * Remove conditional rendering logic and featured products logic in the product grid (it should always render).
  * Remove iteration-specific variations of the Jetpack Free and Jetpack CRM card components.
  * Remove iteration-conditional rendering for the backup storage display component on the My Plan and Backup pages.
  * Remove conditional logic to determine some lists of plans.
  * Remove conditional logic for which plans and products to display in the product grid.
  * Remove conditional logic for product ordering in the product grid.
  * Remove conditional gating around the route for the `/pricing/storage` page.
  * Remove conditional button and body text from the `backup-upsell` and `wpcom-scan-upsell` components.
  * Remove conditional rendering logic from the `wpcom-backup-upsell` component.
  * Remove feature flag gating around date visibility and up-sell logic in all versions of the Activity Log and Backup pages.
  * Remove feature flag gating around storage policy fetching on the Backup and Activity Log pages.
  * Remove the `product-without-price` component entirely, as it is no longer used.
  * Remove unused styles for free products from the `product-grid` component.
  * Remove conditional logic for plan features and copy from the plans list.

* Updates:
  * Move some price-related style rules for the Jetpack Free and Jetpack CRM Free cards into the `display-price` component.

#### Testing instructions

1. Verify there are no longer any references to the following in the wp-calypso repository:
  * `activity-log/display-rules`
  * `jetpack/only-realtime-products`
  * `ONLY_REALTIME_PRODUCTS`
2. Carefully read through all code changes to make sure no obvious issues have been introduced.
3. Regression test the following pages to ensure they look and behave identically to production:
  * Backup
  * Activity Log (in both Calypso Blue and Green)
  * Pricing pages (Calypso Blue, Green, and Jetpack Connect)
